frontend/src/components/pages/RoutePlanningPage.tsx

```diff
@@ -26,32 +37,90 @@ const MapWrapper = () => {
 
   return (
     <div style={style}>
-      <MapView />
+      <MapView
+        onRouteDataChange={onRouteDataChange}
+        initialData={initialData}
+      />
     </div>
   );
 };
 
+interface RouteRecord {
+  id: number;
+  route_name: string;
+  route_data?: string | { points: Point[]; segments: RouteSegment[] };
+}
+
+interface GetRoutesResponse {
+  success: boolean;
+  data?: RouteRecord[];
+}
+
 const RoutePlanningPage = () => {
-  /* id and route_name may be undefined, if the user is creating a new route */
   const [searchParams] = useSearchParams();
   const initialRouteId = parseInt(searchParams.get("id") || "-1");
   const [routeId, setRouteId] = useState<number>(initialRouteId);
```

Comment:
You have already defined an interface called RouteRecord, please do not still define routeId, routeName and routeData separately but use your interface object as a whole for these three entries, AS WELL for initialRouteId, initialRouteName, and initialRouteData.